Inclusion of ACA dataset

ryulippmann opened the pull request:
'b' scripts run ACA dataset analysis

Analysis includes wave data. 
To be functioned out to be commonly included after merging a & b branches.

ryu-AIMS committed:
Included wave prep and analysis

Callaghan's wave data (Hs and Tp) is included in 1b & 2b

ryu-AIMS committed:
Included wave prep and analysis

Callaghan's wave data (Hs and Tp) is included in 1b & 2b

ryu-AIMS committed:
Merge branch 'Wave_Tp_ACA' of https://github.com/ConnectedSystems/GBR-Bathy10m into Wave_Tp_ACA

eco-ben commented on src/1b_prep_aca.jl:
```diff
@@ -68,7 +68,7 @@ proj_str = ProjString(AG.toPROJ4(AG.importWKT(crs(aca_bathy).val; order=:complia
 region_features[!, :geometry] = Vector{AG.IGeometry}(AG.forceto.(region_features.geometry, AG.wkbMultiPolygon))
 
-@showprogress dt = 10 "Prepping benthic/geomorphic/wave data..." for reg in REGIONS
+@showprogress dt = 10 "Prepping bathymetric/turbidity data..." for reg in REGIONS
```
Add /wave to the @showprogress line

eco-ben commented on src/1b_prep_aca.jl:
```diff
@@ -96,4 +96,75 @@ region_features[!, :geometry] = Vector{AG.IGeometry}(AG.forceto.(region_features
+
+    if !isfile(joinpath(ACA_OUTPUT_DIR, "$(reg)_waves_Hs.tif"))
+        target_waves_Hs_path = first(glob("*.nc", joinpath(WAVE_DATA_DIR, "Hs", reg)))
+        target_waves_Hs = Raster(target_waves_Hs_path, key=:Hs90, crs=crs(src_bathy), mappedcrs=EPSG(4326), lazy=true)
```
there isn't a src_bathy raster in the b script. You could use `src_aca_bathy` like in the turbidity .tif processing section before this.

eco-ben commented on src/1b_prep_aca.jl:
```diff
@@ -96,4 +96,75 @@ region_features[!, :geometry] = Vector{AG.IGeometry}(AG.forceto.(region_features
+        target_waves_Hs = Raster(target_waves_Hs_path, key=:Hs90, crs=crs(src_bathy), mappedcrs=EPSG(4326), lazy=true)
+
+        # Extend bounds of wave data to match bathymetry if needed
+        if size(src_bathy) !== size(target_waves_Hs)
```
src_aca_bathy

eco-ben commented on src/1b_prep_aca.jl:
```diff
@@ -96,4 +96,75 @@ region_features[!, :geometry] = Vector{AG.IGeometry}(AG.forceto.(region_features
+            replace_missing!(target_waves_Tp, -9999.0)
+        end
+
+        # Hacky workaround:
```
It might be best to document this comment section in the README if it's not already there so we don't have to repeat it a couple times in a and b scripts.

ryu-AIMS committed:
Updated src variable naming

'src_bathy' renamed as 'src_aca_bathy'
Comment regarding projection workaround for wave data moved to README file

ryu-AIMS committed:
updated iterator name in 1b

missed reg -> region for iterator name

eco-ben commented on src/1b_prep_aca.jl:
```diff
@@ -96,4 +96,61 @@ region_features[!, :geometry] = Vector{AG.IGeometry}(AG.forceto.(region_features
+
+    if !isfile(joinpath(ACA_OUTPUT_DIR, "$(reg)_waves_Hs.tif"))
+        target_waves_Hs_path = first(glob("*.nc", joinpath(WAVE_DATA_DIR, "Hs", reg)))
+        target_waves_Hs = Raster(target_waves_Hs_path, key=:Hs90, crs=crs(src_aca_bathy), mappedcrs=EPSG(4326), lazy=true)
```
I think you will have to define src_aca_bathy after the first if(bathy file) statement, but outside of the turbidity if statement. Because at the moment in the turbidity section src_aca_bathy is removed to nothing.

ryu-AIMS committed:
Variable referencing issue corrected

reigion and region_geom referenced correctly.
src_aca_bathy initialization taken outside of if statement

eco-ben commented on src/1b_prep_aca.jl:
```diff
@@ -96,4 +96,61 @@ region_features[!, :geometry] = Vector{AG.IGeometry}(AG.forceto.(region_features
+            target_waves_Hs = extend(crop(target_waves_Hs; to=src_aca_bathy); to=AG.extent(src_aca_bathy))
+            @assert size(src_aca_bathy) == size(target_waves_Hs)
+
+            replace_missing!(target_waves_Hs, -9999.0)
```
I think you will need to replace -9999.0 with the missing val for aca_bathy. (0 ?)

Commit:
Unable to process wave data for ACA

Wave data requires some reprojection to crs(aca rasters) but we don't know the wave data crs or format.
This won't be as simple as applying the wave data to MPA analysis I think.

ConnectedSystems commented on src/2b_analyse_ACA.jl:
```diff
@@ -36,14 +36,19 @@ include("common.jl")
         # # Source image is of 10m^2 pixels
         # # A hectare is 100x100 meters, so we calculate the proportional area of each hectare
         # that meets criteria (200 <= depth <= 900 cm, and turbidty is no more than LOW (5.2 FNU))
-
+        # 90th percentile of standing wave height is below 1m, and wave period is less than 6 sec).
```
What is FNU?

ConnectedSystems commented:
Uh, I just deleted the `wave-data-analysis` branch as it was merged into `ACA-data-analysis`.

Should this PR target `ACA-data-analysis` or was work not yet finished in that branch?

BG-AIMS commented:
> Uh, I just deleted the `wave-data-analysis` branch as it was merged into `ACA-data-analysis`.
> 
> Should this PR target `ACA-data-analysis` or was work not yet finished in that branch?

I think this PR has been superseded after the ACA data was converted to GDA2020. Maybe it can be closed
